pkg/plugin/plugin.go

```diff
@@ -29,11 +32,33 @@ type Downloaders struct {
 	Command string `json:"command"`
 }
 
-// Plugin interface defines the common methods that all plugin versions must implement
+// Input defined the input message and parameters to be passed to the plugin
+type Input struct {
+	// Message represents the type-elided value to be passed to the plugin
+	// The plugin is expected to interpret the message according to its type/version
+	// The message object must be JSON-serializable
+	Message any
+
+	// Optional: Reader to be consumed plugin's "stdin"
+	Stdin io.Reader
+
+	// Optional: Writers to consume the plugin's "stdout" and "stderr"
+	Stdout, Stderr io.Writer
```

Comment:
was considering stdout could perhaps be io.ReadWriter for postrender plugins. But did you get postrender plugins to work with this as it is now? I see all the postrender tests are commented out.

Reply:
I meant that for postrender we need somewhere to read from and write back to. In the most recent working version on (before #3 was merged) Runtime Postrender gets renderedManifests passed as `*bytes.Buffer` pointer so the value can be modified. We can get into it more when fixing/re-implementing postrenderer plugin type with these interface changes.